### ISSUE TITLE
Cannot minifying CSS with page-margin boxes

### ISSUE THREAD

**Issue by Louis-Tian** [meteor/meteor#7075]
It seems the `standard-minifier-css` doesn't support nested CSS rules. 

style.css ( [relevent CSS3 specification](https://www.w3.org/TR/css3-page/#margin-at-rules)  )

```
@page {
  size: 8.5in 11in;
  margin: 10%;

  @top-left {
    content: "Hamlet";
  }
}
```

error message:

```
=> Errors prevented startup:

   While minifying app stylesheet:
   app/client/style.css:9:3: @page missing '}'

=> Your application has errors. Waiting for file change.
```

meteor version and dependencies (nohting extra, all standard stuff from `meteor create`)
- autopublish            1.0.7  (For prototyping only) Publish the entire database
- blaze-html-templates   1.0.4  Compile HTML templates into reactive UI with Meteo
- ecmascript             0.4.3  Compiler plugin that supports ES2015+ in all .js f
- es5-shim               4.5.10  Shims and polyfills to improve ECMAScript 5 suppo
- insecure               1.0.7  (For prototyping only) Allow all database writes f
- jquery                 1.11.8  Manipulate the DOM using CSS selectors
- meteor-base            1.0.4  Packages that every Meteor app needs
- mobile-experience      1.0.4  Packages for a great mobile user experience
- mongo                  1.1.7  Adaptor for using MongoDB and Minimongo over DDP
- reactive-var           1.0.9  Reactive variable
- standard-minifier-css  1.0.6  Standard css minifier used with Meteor apps by def
- standard-minifier-js   1.0.6  Standard javascript minifiers used with Meteor app
- tracker                1.0.13  Dependency tracker to allow reactive callbacks

**Comment by abernix** [comment]
This is actually getting thrown by the `css-parse` npm module.  There is an existing (relatively old) bug over there at reworkcss/css#23.  Doesn't seem to be a very popular issue though. :(

I would recommend +1'ing that issue if you are experiencing this.  Alternatively, maybe a new, more actively maintained AST CSS parser exists.

**Comment by lydell** [comment]
I’d recommend switching to [postcss](https://github.com/postcss/postcss/).

**Comment by hwillson** [comment]
Since the last update on this issue, PR reworkcss/css#105 was created to address this problem. It hasn't been merged yet unfortunately. Looking for another CSS parsing solution (like [`postcss`](https://github.com/postcss/postcss)) seems like the best long-term course of action, but in the short term we might want to consider forking and maintaining our own version of [reworkcss/css](https://github.com/reworkcss/css). There haven't been any updates made or accepted to the main repo in quite some time, so maintaining a fork should be fairly minimal in terms of effort. This could serve as a stop gap until someone is able to look into replacing the current css parsing approach with another option/library. 

Just a quick FYI - I've briefly tested using `postcss` (via [`postcss-safe-parser`](https://github.com/postcss/postcss-safe-parser) instead of `css-parse` / `css-stringify`, and the parsing (AST generation) does work properly, but there will definitely be some work involved in making the necessary code changes to the [`minifier-css`](https://github.com/meteor/meteor/tree/devel/packages/minifier-css) and [`standard-minifier-css`](https://github.com/meteor/meteor/tree/devel/packages/standard-minifier-css) packages. At a first glance those changes aren't quite trivial, due to the AST structure differences generated by `postcss`.

Let me know what people think about maintaining a fork; I'm happy to tackle that work if it makes sense.

**Comment by vsheyanov** [comment]
What is the best workaround for this issue?

**Comment by mitar** [comment]
Sadly juliancwirko:postcss is not maintained anymore: https://github.com/juliancwirko/meteor-postcss/

I think this just shows how postcss should probably be rolled into the Meteor proper. There is no easy way to do autoprefixing anymore now, and this is an example of an issue with standard CSS minimization because of which I have been preferring postcss.

**Comment by hwillson** [comment]
I should have added a comment here to notify people of PR https://github.com/meteor/meteor/pull/9263. This PR takes the first step of introducing `postcss` into the Meteor CSS build process (to address this issue specifically). If that PR is accepted, we can then start to think about how to open up `postcss` in the build process to allow 3rd party `postcss` plugins.

**Comment by hwillson** [comment]
PR #9263 was merged, so a fix for this should be coming in 1.6.1. Thanks!
